File: app/schemas/email.py
from pydantic import BaseModel


class EmailDraftRequest(BaseModel):
    recipient_name: str
    recipient_email: str
    subject_hint: str | None = None
    purpose: str
    tone: str = "professional"  # "professional" | "friendly" | "formal"
    collections: list[str] | None = None
    n_results: int = 5
    user_id: int | None = None  # fallback when AUTH_ENABLED=False

File: frontend/src/types/email.ts
/** Request body sent to POST /api/v1/email/draft. */
export interface EmailDraftRequest {
  recipient_name: string;
  recipient_email: string;
  subject_hint?: string;
  purpose: string;
  tone?: "professional" | "friendly" | "formal";
  collections?: string[];
  n_results?: number;
  user_id?: number;
}

// SSE events reuse the same types as chat (SSEEvent from types/chat.ts)
